test/celeritas/neutron/NeutronElastic.test.cc

```diff
@@ -0,0 +1,310 @@
+    {
+        XsCalculator calc_micro_xs(shared, MevEnergy{energy});
+        EXPECT_SOFT_EQ(calc_micro_xs(el_id), expected_micro_xs[i]);
+        energy *= factor;
+    }
+
+    // Check the elastic cross section at the upper bound (20 GeV)
+    XsCalculator calc_upper_xs(shared, MevEnergy{2e+4});
+    EXPECT_SOFT_EQ(calc_upper_xs(el_id), 4.67e-23);
+}
+
+TEST_F(NeutronElasticTest, macro_xs)
+{
+    // Calculate the CHIPS elastic neutron-nucleus macroscopic cross section
+    // (\f$ cm^{-1} \f$) in the valid range [1e-5:2000] (MeV)
+    auto material = this->material_track().make_material_view();
+    NeutronElasticMacroXsCalculator calc_xs(model_->host_ref(), material);
+
+    std::vector<real_type> const expected_macro_xs = {0.54527696304096029,
+                                                      0.52957250425960667,
+                                                      0.39293046559628597,
+                                                      1.064289155863954,
+                                                      0.46061008089551958,
+                                                      0.28363890447070578,
+                                                      0.1425404724303469,
+                                                      0.081439808202180136,
+                                                      0.033470862079907314};
+
+    real_type energy = 1e-5;
+    real_type const factor = 1e+1;
+    for (auto i : range(expected_macro_xs.size()))
+    {
+        EXPECT_SOFT_EQ(calc_xs(MevEnergy{energy}), expected_macro_xs[i]);
+        energy *= factor;
+    }
+
+    // Check the CHIPS macroscopic cross section at the upper bound (20 GeV)
+    EXPECT_SOFT_EQ(calc_xs(MevEnergy{2000}), 0.036279681208164501);
+}
+
+TEST_F(NeutronElasticTest, diff_xs_coeffs)
+{
+    // Get A-dependent parameters of CHIPS differential cross sections used
+    // for sampling the momentum transfer.
+    auto const& coeffs = model_->host_ref().coeffs;
+
+    // Set the target isotope: He4 (36 parameters for light nuclei, A < 6.5)
+    ChipsDiffXsCoefficients he4_coeff = coeffs[IsotopeId{1}];
+    EXPECT_EQ(he4_coeff.par.size(), 42);
+    EXPECT_EQ(he4_coeff.par[0], 16000);
+    EXPECT_EQ(he4_coeff.par[10], 26.99741289550769);
+    EXPECT_EQ(he4_coeff.par[20], 0.003);
+    EXPECT_EQ(he4_coeff.par[30], 731.96468887355979);
+    EXPECT_EQ(he4_coeff.par[35], 38680.596799999999);
+    EXPECT_EQ(he4_coeff.par[36], 0);
+
+    // Set the target isotope: Cu63 (42 parameters for heavy nuclei, A > 6.5)
+    ChipsDiffXsCoefficients cu63_coeff = coeffs[IsotopeId{2}];
+    EXPECT_EQ(cu63_coeff.par.size(), 42);
+    EXPECT_EQ(cu63_coeff.par[0], 527.781478797624);
+    EXPECT_EQ(cu63_coeff.par[10], 9.842761904761872);
+    EXPECT_EQ(cu63_coeff.par[20], 4.5646562677427038);
+    EXPECT_EQ(cu63_coeff.par[30], 1984873.0860000001);
+    EXPECT_EQ(cu63_coeff.par[35], 0.15874507866387544);
+    EXPECT_EQ(cu63_coeff.par[41], 7129.2726746278049);
```

Comment:
These should all be `EXPECT_SOFT_EQ`

Reply:
EXPECT_SOFT_EQ  gives an error, 
```
celeritas/src/celeritas/test/testdetail/TestMacrosImpl.hh:131:41: error: static assertion failed: Invalid types for soft equivalence
  131 |     static_assert(can_soft_equiv<VE, VA>(),
      |                   ~~~~~~~~~~~~~~~~~~~~~~^~
...
```

Reply:
Sorry, I meant only the ones with floating point values, lines 151–156, and 141–146.

Reply:
Thanks. I should have paid more attention.  Updated as suggested.